### ISSUE TITLE
Test failures in interior point/mumps interface

### ISSUE THREAD

**Issue by Robbybp** [Pyomo/pyomo#3115]
## Summary

```python
    def test_reallocate_memory_mumps(self):
        # Create a tri-diagonal matrix with small entries on the diagonal
        n = 10000
        small_val = 1e-7
        big_val = 1e2
        irn = []
        jcn = []
        ent = []
        for i in range(n - 1):
            irn.extend([i + 1, i, i])
            jcn.extend([i, i, i + 1])
            ent.extend([big_val, small_val, big_val])
        irn.append(n - 1)
        jcn.append(n - 1)
        ent.append(small_val)
        irn = np.array(irn)
        jcn = np.array(jcn)
        ent = np.array(ent)
    
        matrix = coo_matrix((ent, (irn, jcn)), shape=(n, n))
    
        linear_solver = MumpsInterface()
        linear_solver.do_symbolic_factorization(matrix)
    
        predicted = linear_solver.get_infog(16)
    
        res = linear_solver.do_numeric_factorization(matrix, raise_on_error=False)
>       self.assertEqual(res.status, LinearSolverStatus.not_enough_memory)
E       AssertionError: <LinearSolverStatus.successful: 0> != <LinearSolverStatus.not_enough_memory: 1>

pyomo/contrib/interior_point/linalg/tests/test_realloc.py:48: AssertionError
_________________________ TestReallocation.test_mumps __________________________

self = <pyomo.contrib.interior_point.tests.test_realloc.TestReallocation testMethod=test_mumps>

    @unittest.skipIf(not mumps_available, 'Mumps is not available')
    def test_mumps(self):
        n = 20000
        m = make_model_tri(n, small_val=1e-7)
        interface = InteriorPointInterface(m)
        linear_solver = MumpsInterface()
        # Default memory "buffer" factor: 20
        linear_solver.set_icntl(14, 20)
    
        kkt = interface.evaluate_primal_dual_kkt_matrix()
        res = linear_solver.do_symbolic_factorization(kkt)
        predicted = linear_solver.get_infog(16)
    
        self._test_ip_with_reallocation(linear_solver, interface)
        actual = linear_solver.get_icntl(23)
    
>       self.assertTrue(predicted == 12 or predicted == 11)
E       AssertionError: False is not true
```

What we're trying to test is that the Mumps requires more memory than it initially predicts. Maybe Mumps has improved its memory prediction for this type of matrix (tridiagonal) to realize that it does not need more memory, even if the diagonal is small? Maybe it has increased the "safety factor" that it allocates over the prediction? This seems like something that is difficult to test. A real test for this memory mismatch and reallocation functionality will probably require more thought.

**Comment by blnicho** [comment]
@Robbybp can this be closed now that #3114 has been merged?

**Comment by Robbybp** [comment]
Yes, this can be closed
